.github/workflows/backend-build-and-test.yml

```diff
@@ -0,0 +1,129 @@
+name: Backend build and test
+
+on:
+  push:
+    branches: [master]
+    paths:
+      - 'backend/**'
+      - '!backend/poetry.lock'
+      - '.github/workflows/backend-build-and-test.yml'
+  pull_request:
+    branches: [master]
+    paths:
+      - 'backend/**'
+      - '!backend/poetry.lock'
+      - '.github/workflows/backend-build-and-test.yml'
+
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Login
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD}}
+
+      - name: Build image
+        run: >
+          docker build
+          -t inspirehep/hep:${{ github.sha }}
+          --build-arg LOCK_HASH=${{ hashFiles('backend/poetry.lock') }}
+          backend/
+
+      - name: Push
+        run: docker push inspirehep/hep:${{ github.sha }}
+
+  Integration:
+    runs-on: ubuntu-latest
+    needs: Build
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [integration-async, integration]
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+      postgres:
+        image: postgres:9.6
+        env:
+          POSTGRES_USER: inspirehep
+          POSTGRES_PASSWORD: inspirehep
+          POSTGRES_DB: inspirehep
+        ports:
+        - 5432:5432
+      rabbitmq:
+        image: rabbitmq:3-management
+        ports:
+          - 5672:5672
+      elasticsearch:
+        image: inspirehep/elasticsearch:7.1.1
+        env:
+          bootstrap.memory_lock: true
+          ES_JAVA_OPTS: -Xms1024m -Xmx1024m
+          discovery.type: single-node
+        ports:
+          - 9200:9200
+    steps:
+      - name: Test
+        run: >
+          docker run
+          --entrypoint poetry
+          --network="host"
+          inspirehep/hep:${{ github.sha }}
+          run py.test tests/${{ matrix.suite }}
+
+  Unit:
+    runs-on: ubuntu-latest
+    needs: Build
+    steps:
+      - name: Test
+        run: >
+          docker run
+          --entrypoint poetry
+          --network="host"
+          inspirehep/hep:${{ github.sha }}
+          run py.test tests/unit
+
+  Deploy:
+    runs-on: ubuntu-latest
+    needs: [Build, Integration, Unit]
+    if: ${{ github.event_name == 'push' }}
```

Comment:
I think we need this on `release` as well

Reply:
I will run a different WF for the releases since it's just a matter of re-tagging an existing image.